datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -716,12 +716,29 @@ async fn all_type_literal() -> Result<()> {
             date32_col = arrow_cast('2020-01-01', 'Date32') AND
             binary_col = arrow_cast('binary', 'Binary') AND
             large_binary_col = arrow_cast('large_binary', 'LargeBinary') AND
+            view_binary_col = arrow_cast(arrow_cast('binary_view', 'Binary'), 'BinaryView') AND
```

Comment:
See test `binaryview_type_literal_needs_casting_fix()` below, as for the reason behind the double casting.

Reply:
I removed this workaround in a5bfedd0d